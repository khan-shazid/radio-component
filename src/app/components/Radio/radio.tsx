import React, {useState} from 'react';

import styles from './style.module.css';

import { formatwithComma } from '../../utils';
import RADIO_DEFAULT_IMG from '../../assets/radio.png';
import { StationModel } from '../../models';

interface RadioProps {
    item: StationModel,
    isSelected: boolean,
    handleSelection: (id: any) => void
}

export const Radio = ({ item, isSelected, handleSelection }: RadioProps): JSX.Element => {
  return (
    <li
      data-testid="radio"
      onClick={() => {
          handleSelection(item)
      }}
      className={styles.listItem}>
        <h3 className={styles.listItemTitle}>
            <span data-testid="radio-title-name">{item.name}</span>
            <strong data-testid="radio-title-count">{formatwithComma(item.count)}</strong>
        </h3>

        {isSelected && <div data-testid="radio-selected" className={styles.answer} onClick={(e) => e.stopPropagation()}>
            <div className={styles.plus} onClick={() => console.log("plus clicked")}></div>
            <img src={item.imageUrl ? item.imageUrl : RADIO_DEFAULT_IMG} alt=""/>
            <div className={styles.minus} onClick={() => console.log("minus clicked")}></div>
        </div>}
    </li>
  )
}
