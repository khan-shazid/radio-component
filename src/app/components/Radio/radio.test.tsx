import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import { Radio } from './radio';

afterEach(cleanup);

const STATION = {
  id: 1,
  name: 'Putin FM',
  count: '666',
  imageUrl: ''
};
let handleSelection = jest.fn();
const getComponent = (isSelected = false) => {
  const baseProps = {
    item: STATION,
    isSelected,
    handleSelection
  };
  return <Radio {...baseProps}/>
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponent(), div);
});

it('renders radio title correctly', () => {
    const {getByTestId} = render(getComponent());
    expect(getByTestId('radio-title-name')).toHaveTextContent('Putin FM');
    expect(getByTestId('radio-title-count')).toHaveTextContent('66,6');
});

it('matches with snapshot without selection', () => {
    const tree = renderer.create(getComponent()).toJSON();
    expect(tree).toMatchSnapshot();
});

it('matches with snapshot with selection', () => {
    const tree = renderer.create(getComponent(true)).toJSON();
    expect(tree).toMatchSnapshot();
});

it('calls radio onClick listener', () => {
    const {getByTestId} = render(getComponent());
    fireEvent(
      getByTestId('radio'),
      new MouseEvent('click', {
        bubbles: true,
      }),
    );
    expect(handleSelection).toHaveBeenCalledTimes(1);
    expect(handleSelection).toHaveBeenCalledWith(STATION);
});
