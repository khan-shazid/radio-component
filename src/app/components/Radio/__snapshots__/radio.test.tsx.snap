// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with snapshot with selection 1`] = `
<li
  className="listItem"
  data-testid="radio"
  onClick={[Function]}
>
  <h3
    className="listItemTitle"
  >
    <span
      data-testid="radio-title-name"
    >
      Putin FM
    </span>
    <strong
      data-testid="radio-title-count"
    >
      66,6
    </strong>
  </h3>
  <div
    className="answer"
    data-testid="radio-selected"
    onClick={[Function]}
  >
    <div
      className="plus"
      onClick={[Function]}
    />
    <img
      alt=""
      src="radio.png"
    />
    <div
      className="minus"
      onClick={[Function]}
    />
  </div>
</li>
`;

exports[`matches with snapshot without selection 1`] = `
<li
  className="listItem"
  data-testid="radio"
  onClick={[Function]}
>
  <h3
    className="listItemTitle"
  >
    <span
      data-testid="radio-title-name"
    >
      Putin FM
    </span>
    <strong
      data-testid="radio-title-count"
    >
      66,6
    </strong>
  </h3>
</li>
`;
